### DISCUSSION TITLE
Direct S3 Upload with Public ACL

### DISCUSSION THREAD

**Discussion by deleugpn** [livewire/livewire#4538] [7. Bugs (archived)]
Following on the documentation at https://laravel-livewire.com/docs/2.x/file-uploads#upload-to-s3 I managedo to successfully upload an image directly from Livewire to S3. However, as we can see on https://github.com/livewire/livewire/blob/master/src/WithFileUploads.php#L19 the URL being generated is not passing any ACL parameter and https://github.com/livewire/livewire/blob/master/src/GenerateSignedUploadUrl.php#L17 defaults to `private`. Since I don't have a lot of experience with Livewire, I'm not sure what the best approach would be to override that.
I would rather not have to override the entire `startUpload` method, but I'm not sure if adding a protected method that returns the file visibility to the trait is the best approach.

Are there any recommendations on what I could do to send a pull request for Livewire to allow for Public S3 File Upload?

Thanks!

**Comment by joshhanley** [comment]
@deleugpn yeah I can see the issue. I'm not sure how best to handle that, as you are right, there is no way to change the parameter from private to something else.

**Reply by deleugpn** [reply]
Should I report this as a bug?

**Reply by joshhanley** [reply]
@deleugpn you have already reported it 🙂 if you like, just change the category from help to bug on this discussion.

**Reply by deleugpn** [reply]
Thanks!

**Comment by ymgsmz** [comment]
Nice catch, this bug causes a problem while uploading files to `public` buckets on S3-compatible object storages that don't support object-level ACL, e.g. Backblaze

It generates a URL containing `x-amz-acl=private` that is not acceptable by object storage.

Any plan to address this issue?

**Reply by ymgsmz** [reply]
@joshhanley

**Reply by ymgsmz** [reply]
update: Please find the relevant fix PR at https://github.com/livewire/livewire/pull/9250
